src/main/java/org/broadinstitute/hellbender/tools/exome/eval/EvaluateCopyNumberTriStateCalls.java

```diff
@@ -0,0 +1,772 @@
+                .collect(Collectors.toList()));
+        builder.attribute(GenotypeCopyNumberTriStateSegments.NUMBER_OF_TARGETS_KEY, targets.targetCount(truth));
+        builder.evidence(truth, calls);
+        return builder.make();
+    }
+
+    private Genotype buildAndAnnotateTruthOverlappingGenotype(final String sample, final VariantContext truth, final List<VariantContext> calls,
+                                                              final TargetCollection<Target> targets) {
+        final Genotype truthGenotype = truth.getGenotype(sample);
+        // if there is no truth genotype for that sample, we output the "empty" genotype.
+        if (truthGenotype == null) {
+            return GenotypeBuilder.create(sample, Collections.emptyList());
+        }
+        final int truthCopyNumber = GATKProtectedVariantContextUtils.getAttributeAsInt(truthGenotype,
+                GS_COPY_NUMBER_FORMAT_KEY, truthNeutralCopyNumber);
+        final CopyNumberTriStateAllele truthAllele = copyNumberToTrueAllele(truthCopyNumber);
+
+        final List<Pair<VariantContext, Genotype>> allCalls = calls.stream()
+                .map(vc -> new ImmutablePair<>(vc, vc.getGenotype(sample)))
+                .filter(pair -> pair.getRight() != null)
+                .filter(pair -> GATKProtectedVariantContextUtils.getAttributeAsString(pair.getRight(), GenotypeCopyNumberTriStateSegments.DISCOVERY_KEY,
+                        GenotypeCopyNumberTriStateSegments.DISCOVERY_FALSE).equals(GenotypeCopyNumberTriStateSegments.DISCOVERY_TRUE))
+                .collect(Collectors.toList());
+
+        final List<Pair<VariantContext, Genotype>> qualifiedCalls = composeQualifyingCallsList(targets, allCalls);
+
+        return buildAndAnnotateTruthOverlappingGenotype(sample, targets, truthGenotype, truthCopyNumber,
+                    truthAllele, qualifiedCalls);
+    }
+
+    /**
+     * Reduces a list of calls to the ones that pass filters.
+     * @param targets collection of targets under analysis.
+     * @param calls calls to filter.
+     * @return never {@code null}, but perhaps
+     */
+    private List<Pair<VariantContext, Genotype>> composeQualifyingCallsList(final TargetCollection<Target> targets,
+                                                                            final List<Pair<VariantContext, Genotype>> calls) {
+        return calls.stream()
+                // Filter vc/gt with no concrete call or that the call is the reference.
+                .filter(pair -> !pair.getRight().getAlleles().isEmpty()
+                        && !pair.getRight().getAlleles().get(0).equals(Allele.NO_CALL)
+                        && !pair.getRight().getAlleles().get(0).equals(pair.getLeft().getReference()))
+                // Filter vc/gt with low call quality (SQ).
+                .filter(pair -> {
+                        final double callSQ = callQuality(pair.getRight());
```

Comment:
Conversely, I think you *can* contract this lambda as `pair -> callQuality(pair.getRight()) >= filterArguments.minimumCalledSegmentQuality`.